components/transaction/internal/adapters/rabbitmq/healthcheck.go

```
@@ -0,0 +1,338 @@
+	defer s.mu.RUnlock()
+
+	result := make(map[string]libCircuitBreaker.State, len(s.unhealthyStates))
+	maps.Copy(result, s.unhealthyStates)
+
+	return result
+}
+
+// startRecoveryMonitor starts a goroutine that periodically checks if circuits
+// have been reset (closed) via lib-commons Reset() which doesn't trigger listeners.
+// When a reset is detected, it manually triggers the OnStateChange notification.
+func (s *StateAwareHealthChecker) startRecoveryMonitor() {
```

golangci-lint (2.5.0)

[error] func (*StateAwareHealthChecker).startRecoveryMonitor is unused

(unused)